### PR TITLE
feat(node): filter non-entrypoint Node.js files in `/api` directory

### PR DESCRIPTION
**Pull request by smaeda-ks**
## Summary

- Adds static entrypoint detection for Node.js files in `/api/`, mirroring #14493 which did the same for Python
- Helper/utility files that don't export a valid handler are no longer treated as API routes, avoiding unnecessary Vercel Function creation
- Gated behind `VERCEL_NODE_FILTER_ENTRYPOINTS=1` environment variable for safe incremental rollout

## How it works

A new `isNodeEntrypoint()` function in `@vercel/build-utils` uses regex-based heuristics to detect valid handler export patterns in `.js`, `.mjs`, `.ts`, and `.tsx` files:

| Pattern | Example |
|---------|---------|
| ESM default export | `export default function handler(req, res) {}` |
| CJS module.exports | `module.exports = (req, res) => {}` |
| Named HTTP method exports | `export function GET(request) {}` |
| Fetch export | `export function fetch(request) {}` |
| Re-exports | `export { handler as default }` |
| Server handler | `http.createServer(...)` |

Files that don't match any pattern (e.g., utility functions, type-only files, config exports) are skipped in `maybeGetApiBuilder()` and won't produce a Vercel Function.

## Test plan

- [ ] 34 unit tests for `isNodeEntrypoint()` covering valid entrypoints, non-entrypoints, and edge cases
- [ ] 3 integration tests in fs-detectors verifying builder creation/skipping with actual files on disk
- [ ] Verified all 116+ existing handler files across Node builder and CLI test fixtures match at least one detection pattern
- [ ] Verified feature is inactive without env var (zero behavior change by default)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/build-utils/src/node-entrypoint.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Regex patterns that indicate a file is a valid Node.js API entrypoint.
+ * A file must match at least one pattern to be considered an entrypoint.
+ */
+const VALID_EXPORT_PATTERNS = [
+  // ESM default export: export default function handler() {}
+  /export\s+default\b/,
+  // CJS default export: module.exports = (req, res) => {}
+  /module\.exports\s*=/,
+  // ESM named HTTP method or fetch exports: export function GET() {}
```
Comment (ShayCichocki) [inline packages/build-utils/src/node-entrypoint.ts:16]: 🙏 thank you for the comments for the regex

File: packages/build-utils/src/node-entrypoint.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * let the existing build pipeline handle it.
+ */
+export async function isNodeEntrypoint(
+  file: FileFsRef | { fsPath?: string }
+): Promise<boolean> {
+  try {
+    const fsPath = (file as FileFsRef).fsPath;
+    if (!fsPath) return true;
+    const content = await fs.promises.readFile(fsPath, 'utf-8');
+    if (!content.trim()) return false;
```
Comment (ShayCichocki) [inline packages/build-utils/src/node-entrypoint.ts:56]: Just a curiosity: is there ever a time where an empty file being read could be an error/something failing and this return false would cover it up?

**Review by ShayCichocki — APPROVED** [review]

File: packages/build-utils/src/node-entrypoint.ts (hunk 2 above)
Comment (smaeda-ks) [reply packages/build-utils/src/node-entrypoint.ts:56]: I just tried, but I don't think we throw any errors for empty files today, so it should be fine
